kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiThriftClientCallTimeStatics.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.client
+
+import java.util
+
+import org.apache.hive.service.rpc.thrift.{TGetOperationStatusResp, TOperationHandle, TProtocolVersion, TRowSet, TTableSchema}
+
+import org.apache.kyuubi.operation.FetchOrientation.FetchOrientation
+import org.apache.kyuubi.session.SessionHandle
+
+class KyuubiThriftClientCallTimeStatics(client: KyuubiThriftClient) extends KyuubiThriftClient {
+
+  private var timeCost: Long = 0L
+
+  /**
+   * Get the time cost millis on each call of the function in the [[KyuubiThriftClient]]
+   */
+  def getTimeCostMillis: Long = {
+    timeCost
+  }
+
+  @throws[Exception]
+  private def time[T](f: => T): T = {
+    val startTime = System.currentTimeMillis()
+    try {
+      f
+    } finally {
+      timeCost += System.currentTimeMillis() - startTime
+    }
+  }
+
+  override def openSession(
+      protocol: TProtocolVersion,
+      user: String,
+      password: String,
+      configs: Map[String, String]): SessionHandle = {
+    time(client.openSession(protocol, user, password, configs))
```

Comment:
this is for session RPC, why do we wrap them to track as operation metrics. IIUC, they are not used acturally.